### PR TITLE
Misuv 3961 payments allocation page

### PR DESCRIPTION
**Pull request by SiddharthAtos**

### CODE REVIEW

File: app/models/financialDetails/FinancialDetail.scala [hunk 1]
```diff
@@ -95,14 +96,19 @@ object FinancialDetail {
...
-    for {
-      mainTypeValue <- mainType
-      chargeTypeValue <- chargeType
-      chargeTypeParts <- supportedCTypePartsByMainType.get(mainTypeValue)
-      if chargeTypeParts.exists(supportedCTypePart => chargeTypeValue.startsWith(supportedCTypePart))
-      mainTypeKey <- getMessageKeyForMainType(mainType)
-      chargeTypeKey <- getMessageKeyForChargeType(chargeType)
-    } yield s"$mainTypeKey.$chargeTypeKey"
+    if (MfaDebitUtils.isMFADebitMainType(mainType)) {
+      Option("hmrcAdjustment.text")
```
Comment (joesheridan) [inline app/models/financialDetails/FinancialDetail.scala:100]: Normally we'd put Some("hmrcAdjustment") in place of Option

File: it/testConstants/PaymentAllocationIntegrationTestConstants.scala [hunk 2]
```diff
@@ -317,6 +378,57 @@ object PaymentAllocationIntegrationTestConstants {
 			|}
 			|""".stripMargin)
 
+  val validPaymentAllocationChargesHmrcAdjustmentJson: JsValue = Json.parse(
+    """{
+      |		"documentDetails" : [
+      |			{
+      |       "transactionId":"MA999991A202202",
```
Comment (joesheridan) [inline it/testConstants/PaymentAllocationIntegrationTestConstants.scala:385]: indentation is off here

Comment (SiddharthAtos) [reply it/testConstants/PaymentAllocationIntegrationTestConstants.scala:385]: sure. Will amend it.

File: test/models/financialDetails/FinancialDetailSpec.scala [hunk 3]
```diff
@@ -172,6 +172,7 @@ class FinancialDetailSpec extends UnitSpec {
         getMessageKeyByTypes(Some("SA Balancing Charge"), Some("Voluntary NIC2-GB")) shouldBe Some("bcd.vcnic2")
         getMessageKeyByTypes(Some("SA Balancing Charge"), Some("CGT")) shouldBe Some("bcd.cgt")
         getMessageKeyByTypes(Some("SA Balancing Charge"), Some("SL")) shouldBe Some("bcd.sl")
+        getMessageKeyByTypes(Some("ITSA Misc Charge"), Some("test")) shouldBe Some("hmrcAdjustment.text")
```
Comment (joesheridan) [inline test/models/financialDetails/FinancialDetailSpec.scala:175]: there are 3 other MFA debit strings we could add here

Comment (SiddharthAtos) [reply test/models/financialDetails/FinancialDetailSpec.scala:178]: Yeah, Missed those. 
Thank you.

**Pull request merged**